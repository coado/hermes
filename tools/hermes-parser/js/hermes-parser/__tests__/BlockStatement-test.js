/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict-local
 * @format
 */

import {
  printForSnapshotESTree,
  parseForSnapshotESTree,
  printForSnapshotBabel,
  parseForSnapshotBabel,
} from '../__test_utils__/parse';

describe('BlockStatement', () => {
  describe('implicit block is removed', () => {
    const code = `
    if (a) function a() {}
    `;

    test('ESTree', async () => {
      expect(await parseForSnapshotESTree(code)).toMatchSnapshot();
      expect(await printForSnapshotESTree(code)).toBe(code.trim());
    });

    test('Babel', async () => {
      expect(await parseForSnapshotBabel(code)).toMatchSnapshot();
      expect(await printForSnapshotBabel(code)).toBe(code.trim());
    });
  });
});
