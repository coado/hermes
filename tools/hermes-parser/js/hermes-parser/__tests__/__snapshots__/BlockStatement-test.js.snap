// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockStatement implicit block is removed Babel 1`] = `
{
  "body": [
    {
      "alternate": null,
      "consequent": {
        "async": false,
        "body": {
          "body": [],
          "directives": [],
          "type": "BlockStatement",
        },
        "generator": false,
        "id": {
          "name": "a",
          "type": "Identifier",
        },
        "params": [],
        "type": "FunctionDeclaration",
      },
      "test": {
        "name": "a",
        "type": "Identifier",
      },
      "type": "IfStatement",
    },
  ],
  "type": "Program",
}
`;

exports[`BlockStatement implicit block is removed ESTree 1`] = `
{
  "body": [
    {
      "alternate": null,
      "consequent": {
        "async": false,
        "body": {
          "body": [],
          "type": "BlockStatement",
        },
        "expression": false,
        "generator": false,
        "id": {
          "name": "a",
          "optional": false,
          "type": "Identifier",
          "typeAnnotation": null,
        },
        "params": [],
        "predicate": null,
        "returnType": null,
        "type": "FunctionDeclaration",
        "typeParameters": null,
      },
      "test": {
        "name": "a",
        "optional": false,
        "type": "Identifier",
        "typeAnnotation": null,
      },
      "type": "IfStatement",
    },
  ],
  "type": "Program",
}
`;
